lib/screens/package_details/package_details_event.dart

```diff
@@ -0,0 +1,19 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+
+@immutable
+abstract class PackageDetailsEvent {}
+
+class GetPackageDetailsEvent extends PackageDetailsEvent {
+  final String packageName;
+  final int packageScore;
+  final Completer onComplete = Completer();
```

Comment:
why is this needed? You can just use BlocListener to know when a state change has occurred.

Reply:
The specific custom widget it's used for needs a future and can't get its data from the bloc state,